### DISCUSSION TITLE
`npm ls` should NOT return an exit code of 1 if it finds unmet peer dependencies

### DISCUSSION THREAD

**Discussion by apoorva-sriv** [npm/rfcs#456] [RRFC]
Clone of https://github.com/npm/npm/issues/17624 since that repo was archived
In particular, see [this](https://github.com/npm/npm/issues/17624#issuecomment-313294516) comment.

**Comment by ljharb** [comment]
It absolutely should return nonzero; unmet peer deps mean your entire dep graph is invalid.

**Reply by apoorva-sriv** [reply]
https://github.com/npm/npm/issues/17624#issuecomment-313294516

**Reply by ljharb** [reply]
Then it's misnamed ¯\\\_(ツ)_/¯ that doesn't change that it's working how it's supposed to.

**Reply by apoorva-sriv** [reply]
Well, yes, I wanted to raise this discussion to change that behaviour. Maybe have two separate commands: one to just list the tree (without throwing an error if it's not well formed) and another to check the tree, because throwing an error means the `list` command itself failed, and that's not true, right?

**Reply by ljharb** [reply]
It is true - since its goal is to list the dependency graph, and that's not possible if it's invalid.

**Reply by apoorva-sriv** [reply]
OK, then can the info about the exit code be added to the [`npm ls`](https://docs.npmjs.com/cli/v7/commands/npm-ls) documentation?

**Reply by ljharb** [reply]
That sounds like a great PR! I'm sure the npm team would appreciate it.
